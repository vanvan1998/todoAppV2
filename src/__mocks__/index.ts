const todo = {
  email: 'nguyensivan1998+102@gmail.com',
  id: '82zB2At6gnxs0SbDJ8SW',
  notification: true,
  createdAt: '2024-05-10T01:14:38',
  title: '333333',
  detail: '3333333',
  startTime: '01:00:21',
  startDate: '05/10/2024',
  category: 1,
  completed: false,
  userId: 'j3WklyoTrBg4E1GKNz0Gn0az34V2',
  checkList: []
};
export { todo };
