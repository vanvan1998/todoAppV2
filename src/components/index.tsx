export { AddTodoModal } from './AddTodoModal';
export { Search } from './Search';
export { TodoItem } from './TodoItem';
export { Container } from './Container';
export { Header } from './Header';
export { AddTodo } from './addTodo';
export { SortBy } from './SortBy';
export { Input } from './input';
export { Button } from './button';
