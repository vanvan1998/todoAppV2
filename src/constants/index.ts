export const SORT_BY = {
  NEWEST: 'NEWEST',
  OLDEST: 'OLDEST',
  NONE: 'NONE'
};

export const MODE = {
  UPDATE: 'UPDATE',
  ADD: 'ADD'
};

export const DATE_FORMAT = 'MM/DD/YYYY';
export const TIME_FORMAT = 'HH:mm:ss';
