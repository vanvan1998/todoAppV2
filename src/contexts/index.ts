export * from './AppProvider';
export * from './GlobalLoadingProvider';
export * from './AuthProvider';
