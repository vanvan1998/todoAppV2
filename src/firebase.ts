import { initializeApp } from 'firebase/app';
import { getAuth, signInWithPopup } from 'firebase/auth';
import { getStorage } from 'firebase/storage';
import { getFirestore } from 'firebase/firestore';
import { GoogleAuthProvider } from 'firebase/auth';

export const app = initializeApp({
  apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
  authDomain: process.env.NEXT_PUBLIC_FIREBASE_AUTH_DOMAIN,
  databaseURL: process.env.NEXT_PUBLIC_FIREBASE_DATABASE_URL,
  projectId: process.env.NEXT_PUBLIC_FIREBASE_PROJECT_ID,
  storageBucket: process.env.NEXT_PUBLIC_FIREBASE_STORAGE_BUCKET,
  messagingSenderId: process.env.NEXT_PUBLIC_FIREBASE_MESSAGING_SENDER_ID,
  appId: process.env.NEXT_PUBLIC_FIREBASE_APP_ID
});

const provider = new GoogleAuthProvider();
provider.setCustomParameters({ prompt: 'select_account' });

export const auth = getAuth(app);
export const storage = getStorage(app);
export const db = getFirestore(app);

export const signInWithGoogle = () => signInWithPopup(auth, provider);