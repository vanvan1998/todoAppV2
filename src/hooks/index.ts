export * from './useFirebase';
export * from './useNotification';
export * from './useMediaQuery';
