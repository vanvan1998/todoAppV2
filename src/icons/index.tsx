export * from './AccountIcon';
export * from './HomeIcon';
export * from './CheckedIcon';
export * from './UnCheckedIcon';
export * from './EditIcon';
export * from './DeleteIcon';
export * from './ExpandLessIcon';
export * from './ExpandMoreIcon';
export * from './CloseIcon';
export * from './PersonIcon';
export * from './UrgentIcon';
export * from './ImportantIcon';
export * from './UpcomingIcon';
export * from './SignOutIcon';
