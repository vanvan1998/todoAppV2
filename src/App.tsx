import React from 'react';
import { AuthProvider } from './contexts/AuthContext';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import {
  Signup,
  LeftMenu,
  Login,
  PrivateRoute,
  ForgotPassword,
  UpdatePassword,
  UpdateProfile,
  DashBoard
} from './modules';

function App() {
  return (
    <div style={{ minHeight: '100vh', display: 'flex' }}>
      <Router>
        <AuthProvider>
          <Routes>
            <Route
              path='/update-password'
              element={
                <PrivateRoute>
                  <UpdatePassword />
                </PrivateRoute>
              }
            />
            <Route
              path='/update-profile'
              element={
                <PrivateRoute>
                  <UpdateProfile />
                </PrivateRoute>
              }
            />
            <Route path='/signup' element={<Signup />} />
            <Route path='/login' element={<Login />} />
            <Route path='/forgot-password' element={<ForgotPassword />} />
            <Route
              path='/left-menu'
              element={
                <PrivateRoute>
                  <LeftMenu />
                </PrivateRoute>
              }
            />
            <Route
              path='/'
              element={
                <PrivateRoute>
                  <DashBoard />
                </PrivateRoute>
              }
            />
          </Routes>
        </AuthProvider>
      </Router>
    </div>
  );
}

export default App;
