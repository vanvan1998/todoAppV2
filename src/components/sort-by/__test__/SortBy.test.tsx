import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { SortBy } from '..';
import { SORT_BY } from 'src/constants';

describe('Test SortBy component', () => {
  test.only('should match its snapshot', async () => {
    const component = render(<SortBy sortType={SORT_BY.NEWEST} handleSort={jest.fn()} />);

    const button = component.getByTestId('sort-by-button');
    fireEvent.click(button);
    expect(component).toMatchSnapshot();
  });
});
