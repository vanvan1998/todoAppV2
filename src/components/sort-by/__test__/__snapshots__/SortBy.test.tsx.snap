// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test SortBy component should match its snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-iqyJx ezmOzM"
      >
        <div>
          <button
            buttontype="secondary"
            class="sc-ghWlax a-dPbm"
            data-testid="sort-by-button"
            style="border-radius: 4px; padding: 0px 12px; min-height: 36px;"
            type="button"
          >
            <svg
              fill="#6459e3"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 17H23L19 21L15 17H18V3H20V17M8 5C4.14 5 1 8.13 1 12C1 15.87 4.13 19 8 19C11.86 19 15 15.87 15 12C15 8.13 11.87 5 8 5M8 7.15C10.67 7.15 12.85 9.32 12.85 12C12.85 14.68 10.68 16.85 8 16.85C5.32 16.85 3.15 14.68 3.15 12C3.15 9.32 5.32 7.15 8 7.15M7 9V12.69L10.19 14.53L10.94 13.23L8.5 11.82V9"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-iqyJx ezmOzM"
    >
      <div>
        <button
          buttontype="secondary"
          class="sc-ghWlax a-dPbm"
          data-testid="sort-by-button"
          style="border-radius: 4px; padding: 0px 12px; min-height: 36px;"
          type="button"
        >
          <svg
            fill="#6459e3"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 17H23L19 21L15 17H18V3H20V17M8 5C4.14 5 1 8.13 1 12C1 15.87 4.13 19 8 19C11.86 19 15 15.87 15 12C15 8.13 11.87 5 8 5M8 7.15C10.67 7.15 12.85 9.32 12.85 12C12.85 14.68 10.68 16.85 8 16.85C5.32 16.85 3.15 14.68 3.15 12C3.15 9.32 5.32 7.15 8 7.15M7 9V12.69L10.19 14.53L10.94 13.23L8.5 11.82V9"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
